src/v/kafka/server/request_context.h

```diff
@@ -192,6 +196,8 @@ class request_context {
     request_header _header;
     request_reader _reader;
     ss::lowres_clock::duration _throttle_delay;
+
+    kafka::topic_probe _topic_probe;
```

Comment:
I think the lifetime of request context objects is roughly the same as the lifetime of a request in the system, which means that it will get destroyed when request processing is complete. The `ss::metrics::metric_groups` destructor (owned by the probe) removes all the registered metrics so they're lost after each request.

The probe should probably be stashed somewhere else (to match the lifetime of the server maybe).